### ISSUE TITLE
var-naming: Go stdlib collision check breaks defaulting to golint behavior

### ISSUE THREAD

**Issue by chaimleib** [mgechev/revive#1602]
**Describe the bug**
The `RULES_DESCRIPTIONS.md` promises about the `var-naming` rule:

> By default, the rule behaves exactly as the alternative in golint but optionally, you can relax it (see https://github.com/golang/lint/issues/89).

However, the new v1.13.0 behavior of `skipPackageNameCollisionWithGoStd` conflicts with that promise, making our behavior stricter than the original.

Given the following folder structure:
```text
http/
    http.go
```

And given the following content of `http.go`:
```go
# Package http contains helpers.
package http
```

The original behavior of `golint` gives no errors.

```bash
$ go install golang.org/x/lint/golint@latest
$ cd http
$ golint .
```

**To Reproduce**
1. I updated revive `go install github.com/mgechev/revive@latest`
2. I run it with no flags & no configuration file, inside the `http` folder described above:

```shell
revive .
```

**Expected behavior**
No errors.

**Logs**

> http.go:2:9: avoid package names that conflict with Go standard library package names

**Desktop (please complete the following information):**
 - OS: Fedora 43
 - Version of Go: 1.25.4

**Additional context**
The last version of revive with the expected behavior is v1.12.0.

**Comment by chaimleib** [comment]
I made a PR #1603 to fix this. It deprecates `skipPackageNameCollisionWithGoStd` and inverts its value, if present, to the replacement option `checkPackageNameCollisionWithGoStd`.

**Comment by denisvmedia** [comment]
Won't it make existing setups of revive broken due to backwards incompatible changes in the config?

**Comment by chaimleib** [comment]
I'm working on graceful deprecation logic, so that stuff won't break, despite the config changes.

**Comment by chaimleib** [comment]
@denisvmedia what do you think of the PR now?

**Comment by ldemailly** [comment]
I just got
```
version/version.go:17:9: var-naming: avoid package names that conflict with Go standard library package names (revive)
package version // import "fortio.org/fortio/version"
        ^
```
I am not aware/can't find a stdlib package named `version` 
same error for `metrics` which afaik is also not in stdlib

(should I open a separate bug? meanwhile how do I turn it off?)

**Comment by alexandear** [comment]
@ldemailly thanks for the comment

> I am not aware/can't find a stdlib package named version same error for metrics which afaik is also not in stdlib

From https://pkg.go.dev/std: [`go/version`](https://pkg.go.dev/go/version) and [`runtime/metrics`](https://pkg.go.dev/runtime/metrics).

> meanwhile how do I turn it off?

From https://revive.run/r#var-naming:

```toml
[rule.var-naming]
arguments = [[], [], [{ skip-package-name-collision-with-go-std = true }]]
```

Golangci-lint:

```yaml
linters:
  settings:
    revive:
      enable-default-rules: true
      rules:
        - name: var-naming
          arguments:
           - []
           - []
           - - skip-package-name-collision-with-go-std: true
```

[gitlab-org/api/client-go example](https://gitlab.com/gitlab-org/api/client-go/-/blob/86ac3eaa66087f7144e784967533cda00020591f/.golangci.yml#L71) on disabling this check.

**Comment by ldemailly** [comment]
Thanks, I filled #1637

**Comment by ldemailly** [comment]
@alexandear thanks a lot for the config, somehow that makes it nolintlint/not complain anymore about:
```
periodic/periodic.go:282:6: exported: type name will be used as periodic.PeriodicRunner by other packages, and that stutters; consider calling this Runner (revive)
type PeriodicRunner interface {
     ^
```
any idea how to keep the stutter complaint?

**Comment by ldemailly** [comment]
also
```
jsonschema: "linters.settings.revive" does not validate with "/properties/linters/properties/settings/properties/revive/$ref/additionalProperties": additional properties 'enable-default' not allowed
```

**Comment by ldemailly** [comment]
fixed thanks to ludovic https://github.com/fortio/workflows/pull/195/changes/6ebef1720b70ee1cd18f7e106eb18f808d35eafd : `-rules` was missing in `enable-default-rules`
